### ISSUE TITLE
Row.fetchAll failing

### ISSUE THREAD

**Issue by bibhas2** [groue/GRDB.swift#40]
We are seeing an error with Row.fetchAll. This is happening in two different Macbooks. We are running the app on a simulator.

fatal error: 'try!' expression unexpectedly raised an error: SQLite error 14 with statement `select * from user_profile`: unable to open database file: file /Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-703.0.18.1/src/swift/stdlib/public/core/ErrorType.swift, line 54

```swift
public class func getUserClientId() -> Int? {
    let db = try! DatabasePool(path: getDatabaseFile())
        
    for row in Row.fetchAll(db, "select * from user_profile") {
        let clientId : Int = row.value(named: "client_id")
            
        return clientId
    }
        
    return nil
}

```

```objc
(NSString*) getDatabaseFile {
    NSArray *paths = NSSearchPathForDirectoriesInDomains(
                                        NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *docsDir = [paths objectAtIndex:0];
    NSString *dbFile = [docsDir
                        stringByAppendingPathComponent: @"database.dat"];
    NSLog(@"Database file: %@", dbFile);
    
    return dbFile;
}
```

**Comment by groue** [comment]
Hello @bibhas2.

Your sample code liberally creates instances of DatabasePool. This breaks the Rule 1 of GRDB concurrency:

> Rule 1: Your application should have a unique instance of DatabaseQueue or DatabasePool connected to a database file. You may experience concurrency trouble if you do otherwise.

It's a good time to have a look at https://github.com/groue/GRDB.swift#concurrency.

Tell me if this fixes your issue.

**Comment by bibhas2** [comment]
Creating a database queue instead of a pool fixed the problem.

```swift
let db = try! DatabaseQueue(path: SchemaManager.getDatabaseFile())
```

Thanks!

**Comment by groue** [comment]
Hello @bibhas2. I believe that your issue with DatabasePool has been fixed in v0.79.4 (see #102 for details). If you have time, will you use DatabasePool again, and tell us if the error has gone?
